ironfish/src/consensus/verifier.test.ts

```diff
@@ -436,4 +448,51 @@ describe('Verifier', () => {
       )
     })
   })
+
+  describe('verifyBlockConnect', () => {
+    const nodeTest = createNodeTest()
+
+    it('says a block created before V1 consensus upgrade with double spend is valid', async () => {
+      const { chain, node } = nodeTest
+      const { block } = await useBlockWithTx(node)
+      block.header.sequence = chain.consensus.V1_DOUBLE_SPEND - 1
+
+      const spends = Array.from(block.spends())
+      jest.spyOn(block, 'spends').mockImplementationOnce(function* () {
+        for (const spend of spends) {
+          yield spend
+          yield spend
+        }
+      })
+
+      expect((await chain.verifier.verifyBlockConnect(block)).valid).toBe(true)
+    })
+
+    it('says a block created after V1 consensus upgrade with double spend is invalid with DOUBLE_SPEND as reason', async () => {
+      const { chain, node } = nodeTest
+      const { block } = await useBlockWithTx(node)
+      block.header.sequence = chain.consensus.V1_DOUBLE_SPEND
+
+      const spends = Array.from(block.spends())
+      jest.spyOn(block, 'spends').mockImplementationOnce(function* () {
+        for (const spend of spends) {
+          yield spend
+          yield spend
+        }
+      })
+
+      expect(await chain.verifier.verifyBlockConnect(block)).toEqual({
+        valid: false,
+        reason: VerificationResultReason.DOUBLE_SPEND,
+      })
```

Comment:
I don't think this has anything to do with consensus, because this would always have been caught by our verifier.